zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +148,205 @@ def compute(self, today, assets, out, close, volume):
         out[:] = nanmean(close * volume, axis=0)
 
 
+class CorrelationFactor(SingleInputMixin, CustomFactor):
```

Comment:
I would also be okay with implementing multiple factors each named for their correlation, my only worry is that might be confusing for beginner users, adding to the complexity of pipeline. People are used to thinking about 'correlation', not the underlying type.